### PR TITLE
Optimize inference only mode memory if ipex is used

### PR DESCRIPTION
**Pull request by sywangyi**
Signed-off-by: Wang, Yi A <yi.a.wang@intel.com>

# What does this PR do?

optimize the memory if used ipex for inference.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Library:

- trainer: @sgugger

### CODE REVIEW

**Comment by sywangyi** [comment]
@sgugger  @jianan-gu @yao-matrix please review

File: src/transformers/trainer.py [hunk 1]
```diff
@@ -1309,8 +1309,12 @@ def ipex_optimize_model(self, model, training=False, dtype=torch.float32):
 
         if not training:
             model.eval()
+            inplace = False
+            if not self.is_in_train:
+                dtype = torch.bfloat16 if self.args.bf16_full_eval else dtype
+                inplace = True
```
Comment (sgugger) [inline src/transformers/trainer.py:1315]: Looks like `inplace` is just `not self.is_in_train` so let's just define that, or pass that in the line optimizing the model below.

**Review by sgugger — COMMENTED** [review]
Thanks for your PR! Just left a small comment on the style.

**Review by sgugger — APPROVED** [review]
Much simpler, thanks a lot!

**Pull request merged**